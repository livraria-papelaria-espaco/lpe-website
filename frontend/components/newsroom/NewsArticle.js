import { Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';
import Markdown from '../text/Markdown';

const useStyles = makeStyles((theme) => ({
  root: {
    marginTop: theme.spacing(2),
  },
  markdown: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(3),
  },
}));

const NewsArticle = ({ title, date, content }) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Typography variant='h3' component='h2'>
        {title}
      </Typography>
      <Typography variant='subtitle2' color='textSecondary'>{`Publicado a ${new Date(
        date
      ).toLocaleDateString('pt-PT')}`}</Typography>
      <Markdown className={classes.markdown}>{content}</Markdown>
    </div>
  );
};

NewsArticle.propTypes = {
  title: PropTypes.string,
  date: PropTypes.string,
  content: PropTypes.string,
};

NewsArticle.defaultProps = {
  title: '',
  date: '',
  content: '',
};

export default NewsArticle;
