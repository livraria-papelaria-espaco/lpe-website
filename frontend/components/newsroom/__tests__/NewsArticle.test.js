import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from '../NewsArticle';

describe('NewsArticle (components/newsroom/NewsArticle', () => {
  it('Renders correctly and matches snapshot', () => {
    const title = 'Lorem Ipsum';
    const content = `
Lorem ipsum dolor sit amet, consectetur adipiscing elit.  
Donec non eros dignissim, dignissim erat eget, tincidunt magna.
Ut nisl risus, *posuere* non **finibus quis, consequat vitae** lacus.`;
    const date = new Date('2019-02-19T15:25:00.000Z').toISOString();

    const wrapper = shallow(<NewsArticle title={title} content={content} date={date} />);
    expect(wrapper).toMatchSnapshot();
  });
});
