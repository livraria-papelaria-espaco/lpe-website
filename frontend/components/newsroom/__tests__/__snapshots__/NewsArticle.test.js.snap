// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle (components/newsroom/NewsArticle Renders correctly and matches snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Typography))
    component="h2"
    variant="h3"
  >
    Lorem Ipsum
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    color="textSecondary"
    variant="subtitle2"
  >
    Publicado a 19/02/2019
  </WithStyles(ForwardRef(Typography))>
  <Markdown
    className="makeStyles-markdown-2"
  >
    
Lorem ipsum dolor sit amet, consectetur adipiscing elit.  
Donec non eros dignissim, dignissim erat eget, tincidunt magna.
Ut nisl risus, *posuere* non **finibus quis, consequat vitae** lacus.
  </Markdown>
</div>
`;
